src/config/model.c

```diff
@@ -29,8 +29,41 @@ struct Model Model;
 static u32 crc32;
 
 const char * const MODEL_TYPE_VAL[MODELTYPE_LAST] = { "heli", "plane", "multi" };
-const char * const RADIO_TX_POWER_VAL[TXPOWER_LAST] =
-     { "100uW", "300uW", "1mW", "3mW", "10mW", "30mW", "100mW", "150mW" };
+const u8 RADIO_TX_POWER_COUNT[TX_MODULE_LAST] = {  // number of power settings
+     8,    //   CYRF6936,
+     7,    //   A7105,
+     8,    //   CC2500,
+     4,    //   NRF24L01,
+     8,    //   MULTIMOD,
+     4,    //   R9M (FCC/EU),
+};
+
+char * radio_tx_power_val(enum Radio radio, enum TxPower power) {
+    // entries in this list must match order in enum TxPower definition
+    static const u16 std_powers[TXPOWER_LAST] = { 100, 300, 1, 3, 10, 30, 100, 150};
+    static const u16 nrf_powers[] = { 1, 6, 25, 100};
+    static char * const r9m_powers[] = { "10/25mW", "100/25mW", "500/500", "Auto/200" };
+
+    if (radio >= TX_MODULE_LAST) return NULL;
+    if (power >= RADIO_TX_POWER_COUNT[radio]) power = RADIO_TX_POWER_COUNT[radio]-1;
+
+    switch (radio) {
+    case CYRF6936:
+    case CC2500:
+    case A7105:
+    case MULTIMOD:
+        snprintf(tempstring, TEMPSTRINGLENGTH, "%d%cW", std_powers[power], (power < 2)?'u':'m');
```

Comment:
This is pretty ugly.  Is it really less rom to do this than to just index the list of strings for this case?

Reply:
Below are numbers from devo10 build.  The "reduced strings" stores each string only once.  I'm surprised snprintf doesn't take more rom.

```
Full strings
ROM: 0x08004000 - 0x0803bae4 = 222.72kB
RAM: 0x20000000 - 0x20008d8c =  35.39kB

Reduced strings
ROM: 0x08004000 - 0x0803ba84 = 222.63kB
RAM: 0x20000000 - 0x20008d8c =  35.39kB

Numbers (sprintf)
ROM: 0x08004000 - 0x0803ba74 = 222.61kB
RAM: 0x20000000 - 0x20008d8c =  35.39kB

Numbers (snprintf)
ROM: 0x08004000 - 0x0803ba80 = 222.62kB
RAM: 0x20000000 - 0x20008d8c =  35.39kB
```

Reply:
I'm willing to spend 16 bytes for the nicer code.  Lets do the reduced strings version

Reply:
Reverted to reduced strings code.